pkg/web_app/lib/src/foldable.dart

```diff
@@ -65,30 +51,31 @@ void _setEventForFoldable() {
 
         /// Scroll the smaller amount of the two.
         scrollDiff = max(0, min(screenLimit, outsideViewDiff));
-      }
 
-      /// Do not scroll if the difference is small, otherwise scroll in small
-      /// steps synchronized to the animation frames.
-      if (scrollDiff > 8) {
-        final originalScrollTop = scrollContainer!.scrollTop;
-        final maxSteps = 20;
-        for (var i = 1; i <= maxSteps; i++) {
-          if (i > 1) await window.animationFrame;
-          final nextPos = originalScrollTop + (scrollDiff * i / maxSteps);
-          scrollContainer.scrollTo(0, nextPos);
+        /// Do not scroll if the difference is small, otherwise scroll in small
+        /// steps synchronized to the animation frames.
+        if (scrollDiff > 8) {
+          final originalScrollTop = scrollContainer.scrollTop;
+          final maxSteps = 20;
+          for (var i = 1; i <= maxSteps; i++) {
+            if (i > 1) await window.animationFrame;
+            final nextPos = originalScrollTop + (scrollDiff * i / maxSteps);
+            scrollContainer.scrollTo(0, nextPos);
+          }
         }
       }
-
-      // Wait one animation frame before enabling the content to resize on its own.
-      await window.animationFrame;
-      content.style.maxHeight = 'none';
     }
 
+    h.querySelector('.foldable-icon')!.attributes['tabindex'] = '0';
+
     // listen on trigger events
     h.onClick.listen((e) async {
-      // Toggle state.
-      final isActive = foldable.classes.toggle('-active');
-      await update(isActive);
+      e.preventDefault();
+      await toggle();
+    });
+    h.onKeyDown.where((e) => e.key == 'Enter').listen((e) async {
```

Comment:
Perhaps also spacebar?

Reply:
I was wondering about this, whether we should add `Space` or not, and if we add it here, we should add it at other places too. Created a new issue to track it: https://github.com/dart-lang/pub-dev/issues/7341